UnityProject/Assets/Scripts/Items/Devices/AccessRemote.cs

```diff
@@ -0,0 +1,135 @@
+using System;
+using Doors;
+using Doors.Modules;
+using Mirror;
+using UnityEngine;
+
+
+namespace Items.Devices
+{
+	/// <summary>
+	/// An item that controls department doors remotely.
+	/// </summary>
+	public class AccessRemote : NetworkBehaviour, IInteractable<HandActivate>, ICheckedInteractable<HandApply>
+	{
+		[SyncVar(hook = nameof(SyncCurrentRemoteState))]
+		private AccessRemoteState currentState;
+
+		private SpriteHandler spriteHandler;
+
+		[SerializeField] private Access access;
+		[SerializeField] private SpriteDataSO departmentSprite;
+
+		private void Start()
+		{
+			spriteHandler = GetComponentInChildren<SpriteHandler>();
+			if (spriteHandler == null)
+			{
+				Logger.LogError("[AccessRemote] - Cannot find sprite handler! did you accidentally remove it from this item's children?");
+				return;
+			}
+
+			if (departmentSprite == null)
+			{
+				Logger.LogWarning("[AccessRemote] - No department sprite found, using default sprite instead. (default sprite could be blank however!)");
+				return;
+			}
+			spriteHandler.SetSpriteSO(departmentSprite);
+		}
+
+		private enum AccessRemoteState
+		{
+			Open,
+			Bolts,
+			Emergency
+		}
+
+		public void ServerPerformInteraction(HandActivate interaction)
+		{
+			switch (currentState)
+			{
+				case AccessRemoteState.Open:
+					SyncCurrentRemoteState(currentState, AccessRemoteState.Bolts);
+					break;
+				case AccessRemoteState.Bolts:
+					SyncCurrentRemoteState(currentState, AccessRemoteState.Emergency);
+					break;
+				case AccessRemoteState.Emergency:
+					SyncCurrentRemoteState(currentState, AccessRemoteState.Open);
+					break;
+			}
+			Chat.AddExamineMsg(interaction.Performer, $"Remote mode is set to: {currentState.ToString()}.");
+		}
+
+		public bool WillInteract(HandApply interaction, NetworkSide side)
+		{
+			if (interaction.IsHighlight || interaction.IsAltClick)
+				return false;
+			if (Validations.HasComponent<DoorMasterController>(interaction.TargetObject) == false)
+				return false;
+
```

Comment:
Need this as well:

if (DefaultWillInteract.Default(interaction, side) == false) return false;

Reply:
Remotes are meant to be used from a distance so I'm not adding a distance check.

Reply:
use Validations.CanApply() then as that has a range parameter

Reply:
Done